src/com/facebook/buck/ide/intellij/IjModuleGraphFactory.java

```diff
@@ -315,6 +325,36 @@ public static IjModuleGraph from(
     return new IjModuleGraph(depsBuilder.build());
   }
 
+  private static IjModule createExtraModuleForCompilerOutput(
+      IjModule module, IjProjectConfig projectConfig, ProjectFilesystem projectFilesystem) {
+    Path projectRootPath = Paths.get(projectConfig.getProjectRoot());
+    Path extraModuleBasePath = projectRootPath
+        .resolve("buck-out")
+        .resolve("extra-intellij-output")
+        .resolve(projectRootPath.relativize(module.getModuleBasePath()));
+
+    Path extraModuleRelativePath = projectRootPath.relativize(extraModuleBasePath.normalize());
+    if (!projectFilesystem.exists(extraModuleRelativePath)) {
+      try {
+        projectFilesystem.mkdirs(extraModuleRelativePath);
```

Comment:
We are adding synthetic modules to the module graph here. In order for the module to be properly created and added to the graph, it's module root must be created (since there is no actual code there already). This depends upon knowledge of each module, so it can't happen before the main logic of buck project. Attached is what the module structure would look like for the uber/okbuck sample app. The directories that it would create are listed below. They would only be created for Android libraries and if the project configuration flag enabled them.
- buck-out/extra-intellij-output/another-app
- buck-out/extra-intellij-output/libraries/common
- buck-out/extra-intellij-output/libraries/empty-library

![screen shot 2017-08-08 at 2 51 11 pm](https://user-images.githubusercontent.com/3178606/29096882-d10ec1d0-7c4b-11e7-9fb7-b5d04fc61696.png)